modules/standard/IO.chpl

```diff
@@ -2972,6 +2976,10 @@ pragma "no doc"
 proc file.readerHelper(param kind=iokind.dynamic, param locking=true,
                        region: range(?) = 0.., hints = ioHintSet.empty,
                        style:iostyleInternal = this._style): channel(false, kind, locking) throws {
+  if (region.hasLowBound() && region.low < 0) {
+    throw new IllegalArgumentError("region", "cannot index into a file before byte 0");
```

Comment:
I think the verb "index" here doesn't quite fit, at least to my mind. Maybe something like:
```
file region's lowest accepted bound is 0
```
?

Reply:
A couple of other things (sorry!):
- what happens if low > high?
- Are strides other than "1" supported? Should we mention that somewhere?

Reply:
It segfaults - it also segfaulted before this PR, so I think I'll file that as a future and maybe we'll be able to get to it before 2.0.

Strides other than 1 are basically just ignored, but it'd probably be good to warn about that, especially since specifying a stride wouldn't have been possible before.  I can see us expanding to support other strides in the future, but definitely not before 2.0.

Nothing to be sorry about, you're keeping me honest and these are things the implementation should take into account :)

Reply:
I think the 'stride' thing could just be a documentation ``.. warning::`` for the time being, if that makes it easier. If we go that route it would eventually be nice to see an IllegalArgumentError for a negative stride. I feel like that's the most likely thing that a user would try, and it would be confusing if that didn't work.

Reply:
Yeah, that seems like a reasonable route.  I'd prefer to have checks in the code, but it's starting to feel like scope creep with feature freeze around the corner